app/controllers/application_controller.rb

```diff
@@ -1,2 +1,34 @@
 class ApplicationController < ActionController::Base
+  private
+
+  def authenticate_user!
```

Comment:
I dont think this should be in private method if it is going to be used outside this controller